reader-v2.go

```diff
@@ -180,6 +180,40 @@ func (r *decReaderV20) Read(p []byte) (n int, err error) {
 	return n, nil
 }
 
+// decryptBufferV20 will append plaintext to dst and return the result.
+func decryptBufferV20(dst, src []byte, config *Config) ([]byte, error) {
+	ad, err := newAuthDecV20(config)
+	if err != nil {
+		return nil, err
+	}
+	for len(src) > 0 {
+		buffer := packageV20(src)
+		// Truncate to max package size
+		if len(buffer) > maxPackageSize {
+			buffer = buffer[:maxPackageSize]
+		}
+
+		// Make space in dst
+		payloadLen := buffer.Header().Length()
+		if cap(dst) <= len(dst)+payloadLen {
```

Comment:
```suggestion
		if cap(dst) >= len(dst)+payloadLen {
```
Or am I reading this wrong?!
Example:
```
10 == cap(dst)
 5 == len(dst)
 6 == payloadLen
_______________
if 10 <= 5 + 6 {
    dst = dst[:5+6]  // causes panic due to out-of-range 
} 
```

Reply:
lol. yeah. Fixed and it will now fail tests as well. @aead Updated.